biorxiv/corpora_comparison/figure_generator.py

```diff
@@ -0,0 +1,409 @@
+    plot_pointplot,
+    plot_bargraph,
+    plot_point_bar_figure
+)
+
+
+# In[2]:
+
+
+subset = 20
+
+
+# # KL Divergence Graph
+
+# In[3]:
+
+
+kl_divergence_df = pd.read_csv(
+    "output/comparison_stats/corpora_kl_divergence.tsv", 
+    sep="\t"
+)
+kl_divergence_df.head()
+
+
+# In[4]:
+
+
+g = (
+    p9.ggplot(
+        kl_divergence_df.replace({
+        "biorxiv_vs_nytac":"biorxiv_vs_reference",
+        "pmc_nytac": "pmc_vs_reference"
+        })
+    )
+    + p9.aes(x="factor(num_terms)", y="KL_divergence", fill="comparison")
+    + p9.geom_col(stat='identity', position='dodge')
+    + p9.scale_fill_manual(["#b2df8a", "#1f78b4", "#a6cee3"])
+    + p9.labs(
+        x = "Number of terms evaluated",
+        y = "KL Divergence"
+    )
+    + p9.theme_seaborn(context='paper')
+)
+g.save("output/figures/corpora_kl_divergence.png", dpi=500)
+print(g)
+
+
+# # bioRxiv vs Pubmed Central
+
+# In[5]:
+
+
+full_text_comparison = pd.read_csv(
+    "output/comparison_stats/biorxiv_vs_pmc_comparison.tsv",
+    sep="\t"
+)
+full_text_comparison.head()
+
+
+# ## Line Plots
+
+# In[6]:
+
+
+full_plot_df = calculate_confidence_intervals(
+    full_text_comparison
+)
+full_plot_df.head()
+
+
+# In[7]:
+
+
+plot_df = (
+    full_plot_df
+    .sort_values("odds_ratio", ascending=False)
+    .head(subset)
+    .append(
+        full_plot_df
+        .sort_values("odds_ratio", ascending=False)
+        .iloc[:-2]
+        .tail(subset)
+    )
+    .replace('rna', 'RNA')
+)
+plot_df.head()
+
+
+# In[8]:
+
+
+g = plot_pointplot(plot_df, "bioRxiv vs PMC Odds Ratio")
+g.save("output/svg_files/biorxiv_pmc_frequency_odds.svg", dpi=75)
+print(g)
+
+
+# In[9]:
+
+
+count_plot_df = create_lemma_count_df(plot_df, 'bioRxiv', 'pmc')
+count_plot_df.head()
+
+
+# In[10]:
+
+
+g = plot_bargraph(count_plot_df, plot_df)
+g.save("output/svg_files/biorxiv_pmc_frequency_bar.svg", dpi=75)
+print(g)
+
+
+# In[11]:
+
+
+fig_output_path = "output/figures/biorxiv_vs_pubmed_central.png"
+
+fig = plot_point_bar_figure(
+    "output/svg_files/biorxiv_pmc_frequency_odds.svg",
+    "output/svg_files/biorxiv_pmc_frequency_bar.svg"
+)
+
+# save generated SVG files
+svg2png(
+    bytestring=fig.to_str(), 
+    write_to=fig_output_path,
+    dpi=500
+)
+
+Image(fig_output_path)
+
+
+# # bioRxiv vs Reference
+
+# In[12]:
+
+
+full_text_comparison = pd.read_csv(
+    "output/comparison_stats/biorxiv_nytac_comparison.tsv",
+    sep="\t"
+)
+full_text_comparison.head()
+
+
+# ## Line Plots
+
+# In[13]:
+
+
+full_plot_df = calculate_confidence_intervals(
+    full_text_comparison
+)
+full_plot_df.head()
+
+
+# In[14]:
+
+
+plot_df = (
+    full_plot_df
+    .sort_values("odds_ratio", ascending=False)
+    .head(subset)
+    .append(
+        full_plot_df
+        .sort_values("odds_ratio", ascending=False)
+        .iloc[:-2]
+        .tail(subset)
+    )
+    .replace('rna', 'RNA')
+)
+plot_df.head()
+
+
+# In[15]:
+
+
+g = plot_pointplot(plot_df, "bioRxiv vs Reference Odds Ratio", True)
```

Comment:
Looks like notation has a higher odds in bioriv and pmc vs reference...that i think makes sense to me since they are writing about different topics. Was that your logic?

Reply:
>Looks like notation has a higher odds in bioriv and pmc vs reference...that i think makes sense to me since they are writing about different topics. Was that your logic?

Yup, you got it. I'd be a bit surprised otherwise.